createFolders.sh

```diff
@@ -1,18 +1,12 @@
 #!/bin/bash
 
-PROJECT_NAME=data-and-algorithms
-DIRECTORY=${PWD}
-BASE_DIRECTORY=$(echo "$DIRECTORY" | rev | cut -d "/" -f1 | rev)
+DIRECTORY="$(dirname "$(readlink "$0")")"
 
-if [[("$1" != "")]]; then
-   if [[("$BASE_DIRECTORY" != "$PROJECT_NAME")]]; then
-      echo "Make sure you are in $PROJECT_NAME project"
-      exit
-   fi
-   rootfileName=$1
-   mkdir ./$rootfileName
-   cp -a ./Python/ ./$rootfileName/
-   rm ./$rootfileName/*/*/*.py
-else
-   echo "Missing folder name: sh createFolders.sh <folder_name>"
-fi
\ No newline at end of file
+if [[ $# -lt 1 ]]; then
+  echo "Usage: bash createFolders.sh <programming language>" >&2
+  exit 1
+fi
+
+programmingLanguage=$1
```

Comment:
nit: remember the double quotes or you'll get weird results if somebody decides to pass in spaces within an argument. https://google.github.io/styleguide/shell.xml?showone=Quoting#Quoting

Rule of thumb is: every single var should be double quoted AND braced (except for the bash builtin ones, so `$1` and `$#` are fine. https://google.github.io/styleguide/shell.xml?showone=Variable_expansion#Variable_expansion

and in general read that whole style guide, it's great!

Reply:
I'll stay consistent and add the double quotes 💯